src/test/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensionSpec.scala

```diff
@@ -0,0 +1,153 @@
+      added.foreach { email => when(mockGoogleDirectoryDAO.addMemberToGroup(target.email, WorkbenchUserEmail(email))).thenReturn(Future.successful(())) }
+      removed.foreach { email => when(mockGoogleDirectoryDAO.removeMemberFromGroup(target.email, WorkbenchUserEmail(email))).thenReturn(Future.successful(())) }
+
+      val addException = new Exception("addError")
+      when(mockGoogleDirectoryDAO.addMemberToGroup(target.email, WorkbenchUserEmail(ge.toProxyFromUser(addError.value)))).thenReturn(Future.failed(addException))
+
+      val removeException = new Exception("removeError")
+      when(mockGoogleDirectoryDAO.removeMemberFromGroup(target.email, WorkbenchUserEmail(removeError))).thenReturn(Future.failed(removeException))
+
+      val results = runAndWait(ge.synchronizeGroupMembers(target.id))
+
+      results.groupEmail should equal(target.email)
+      results.items should contain theSameElementsAs (
+        added.map(SyncReportItem("added", _, None)) ++
+          removed.map(SyncReportItem("removed", _, None)) ++
+          Seq(
+            SyncReportItem("added", ge.toProxyFromUser(addError.value), Option(ErrorReport(addException))),
+            SyncReportItem("removed", removeError, Option(ErrorReport(removeException)))))
+
+      added.foreach { email => verify(mockGoogleDirectoryDAO).addMemberToGroup(target.email, WorkbenchUserEmail(email)) }
+      removed.foreach { email => verify(mockGoogleDirectoryDAO).removeMemberFromGroup(target.email, WorkbenchUserEmail(email)) }
+      verify(mockDirectoryDAO).updateSynchronizedDate(target.id)
+    }
+  }
+
+  "GoogleExtension" should "get a pet service account for a user" in {
+    implicit val patienceConfig = PatienceConfig(1 second)
+    val dirDAO = new JndiDirectoryDAO(directoryConfig)
+    val schemaDao = new JndiSchemaDAO(directoryConfig)
+    runAndWait(schemaDao.init())
+    try {
+      val mockGoogleIamDAO = new MockGoogleIamDAO
+      val mockGoogleDirectoryDAO = new MockGoogleDirectoryDAO
+
+      val googleExtensions = new GoogleExtensions(dirDAO, null, mockGoogleDirectoryDAO, null, mockGoogleIamDAO, googleServicesConfig, petServiceAccountConfig)
+      val service = new UserService(dirDAO, googleExtensions, mockGoogleDirectoryDAO, googleServicesConfig.appsDomain)
+
+      val defaultUserId = WorkbenchUserId("newuser")
+      val defaultUserEmail = WorkbenchUserEmail("newuser@new.com")
+      val defaultUser = WorkbenchUser(defaultUserId, defaultUserEmail)
+
+      // create a user
+      val newUser = service.createUser(defaultUser).futureValue
+      newUser shouldBe UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> true, "allUsersGroup" -> true, "google" -> true))
+
+      // create a pet service account
+      val emailResponse = googleExtensions.createUserPetServiceAccount(defaultUser).futureValue
+
+      emailResponse.value should endWith("@my-pet-project.iam.gserviceaccount.com")
```

Comment:
I wish the `my-pet-project` part of the setup wasn't so far from the test (tucked away in `reference.conf`). Is there a way to make the test more self-contained, maybe by manually constructing `petServiceAccountConfig` instead of having it read from a config file?

Reply:
I could, it is just a PITA... I think I will change the verification to pull the project from config

Reply:
Ah, that works too. Good idea!